docs/recent-releases/go-ipfs-0-5/features.md

```diff
@@ -0,0 +1,160 @@
+- A Libp2p event bus for processing updates like these.
+- A new DHT protocol version. New DHT nodes will not admit old DHT nodes into their routing tables. Old DHT nodes will still be able to issue queries against the new DHT, but they won't be queried or referred by new DHT nodes. This way, old, potentially unreachable nodes with bad routing tables won't pollute the new DHT.
+
+There is a significant downside to this approach, however. Nodes behind VPNs, offline LANs, etc. are not publicly reachable. To address this issue, Go-IPFS 0.5.0 will run two DHTs: one for private networks and one for the public internet. Every node will participate in a LAN DHT and a public WAN DHT. See [Dual DHT](dual-dht) for more details.
+
+#### Dual DHT
+
+All IPFS nodes will now run two DHTs: one for the public internet WAN, and one for their local network LAN.
+
+1. When connected to the public internet, IPFS will use both DHTs for finding peers, content, and IPNS records. Nodes only publish provider and IPNS records to the WAN DHT to avoid flooding the local network.
+2. When not connected to the public internet, nodes publish provider and IPNS records to the LAN DHT.
+
+The WAN DHT includes all peers with at least one public IP address. This release will only consider an IPv6 address public if it is in the [public internet range `2000::/3`](https://www.iana.org/assignments/ipv6-address-space/ipv6-address-space.xhtml).
+
+This feature should not have any noticeable impact on Go-IPFS, performance, or otherwise. Everything should continue to work in all the currently supported network configurations: VPNs, disconnected LANs, public internet, etc.
+
+#### Query Logic
+
+We've fixed the DHT query logic by correctly implementing Kademlia. This should significantly speed up:
+
+- Publishing IPNS & provider records. We previously continued searching for closer and closer peers to the "target" until we timed out, then we put to the closest peers we found.
+- Resolving IPNS addresses. We previously continued IPNS record searches until we ran out of peers to query, timed out, or found 16 records.
+
+In both cases, nodes continue searching until they find the closest peers and then stop.
+
+#### Routing Tables
+
+Finally, we've addressed the poorly maintained routing tables by:
+
+- Reducing the likelihood that the connection manager will kill connections to peers in the routing table.
+- Keeping peers in the routing table, even if we get disconnected from them.
+- Actively and frequently querying the DHT to keep our routing table full.
+
+### Testing
+
+The DHT rewrite was made possible by [Testground](https://github.com/ipfs/testground/), our new testing framework. Testground allows us to spin up multi-thousand node tests with simulated real-world network conditions. By combining Testground and some custom analysis tools, we were able to gain confidence that the new DHT implementation behaves correctly.
+
+## AutoNAT
```

Comment:
Autonat is not new - but outside of the autorelay subsystem wasn't used. perhaps it could be framed as

```
This release uses Automatic NAT Detection (AutoNAT) - determining if the node is _reachable_ from the public internet - to make decisions about how to participate in IPFS. This subsystem is used to determine if the node should store some of the public DHT, and if it needs to use relays to be reached by others.
```